flink-runtime/src/main/java/org/apache/flink/runtime/taskmanager/Task.java

```diff
@@ -1574,6 +1574,12 @@ public void run() {
                 // we need do continue despite that
                 try {
                     invokable.cancel();
+                    // Wait for any active actions to complete (e.g. timers, mailbox actions)
+                    // Before that, interrupt to notify them know about cancellation
+                    if (invokable.shouldInterruptOnCancel()) {
+                        executer.interrupt();
+                    }
+                    invokable.getTerminationFuture().join();
```

Comment:
Moving it to the inner try-catch block changes the log messages a bit. Could we move back the interruption to the outside try-catch block, but before the `networkResourcesCloser`?

Reply:
The interruption must happen before joining the terminationFuture. Otherwise, `SourceStreamTask` will not finish.

Reply:
I see. Ok, that makes sense.